### DISCUSSION TITLE
Navigation Using Arrows Keys

### DISCUSSION THREAD

**Discussion by hero-clashes** [mikke89/RmlUi#280] [Q&A] [answered]
so I have this setup in my game 
```html
<div class="container">
<button type="button">Play</button><br />
<button type="button">Load Game</button><br />
<button type="button">Testing Areana</button><br />
<button type="button">Setting</button><br />
<button type="button">Quit</button><br />
</div>
```
and I want to do navigation using arrows keys between them 
so if I pressed up it go to the button above it and so on
Also if I pressed enter, it would call the activate 
is there a way to archive that?

**Comment by mikke89** [comment, marked as answer]
Hi.

You just need to script this behavior as you see fit. 

- First you need some way to track the state of the currently selected item. For example, you can use a custom class name, or perhaps focus the active element using `element->Focus()`. 
- Then when you register an arrow down click, you need to find the next element. If they are always ordered next to each other in RML, then it is simply `focused_element->GetNextSibling()->Focus()`. Of course you also need end of list and error handling.
- You can do this entirely on the client side, or you can setup event listeners such as to `keydown` events and act when that is triggered. See the [event documentation](https://mikke89.github.io/RmlUiDoc/pages/cpp_manual/events.html) for more details.
- As an extension to the above, you could also use [inline events](https://mikke89.github.io/RmlUiDoc/pages/cpp_manual/events.html#inline-events), eg. first by defining `<button type="button" onkeydown="up testing-arena; down quit">Setting</button>`. Then when it is triggered you parse that little command and choose the appropriate action.  See the _Invader_ sample and how this is done there.

Good luck :)

**Reply by hero-clashes** [reply]
Thanks for the detailed answer, I will try one of the suggestions and reply
Hope you have good day and good luck
Btw: is there is any interest of Raylib rendering backend, I made it for my game, but I can send a pull request

**Reply by mikke89** [reply]
Great, I'm always glad that you want to contribute. Please do post a PR if you'd like! I will only merge the PR if it is a backend that we want to support long-term, but whether or not we merge it I'm sure it will be useful to others. Otherwise, if the backend is available somewhere online, I can also add a link to #186.

**Reply by hero-clashes** [reply]
so it worked, I did all input handling outside Rml, though I still excepted it to do that for me as most UI need some sort of keyboard controls but no problem
thanks again for the help

**Reply by mikke89** [reply]
I'm glad you got it working! There is a topic for built-in controller support, see #142.

**Reply by hero-clashes** [reply]
I found an Issue I feed the Rml a string which is something like this "┼" and it show a box like it didn't know the char, any way I can get support for those chars, btw I only use chars from [this](http://dwarffortresswiki.org/index.php/Character_table)

**Reply by mikke89** [reply]
You need to feed RmlUi a font which actually contains the character you want to show.

**Reply by hero-clashes** [reply]
you are right, thanks again for third time
hope you have great day

**Reply by hero-clashes** [reply]
@mikke89  is there any way I can contact you in something like discord, I am interested in tackling the system for navigation mentioned in [this](https://github.com/mikke89/RmlUi/issues/142) and I probably need some guidance on how to do stuff and so on, if you are interested

**Reply by mikke89** [reply]
Sure, we have a [gitter chat](https://gitter.im/RmlUi/community), feel free to post your questions there!
